app/assets/main/react_components/footprint_form/FootprintForm.jsx

```diff
@@ -130,72 +78,36 @@ const FootprintForm = ({ calculator, questionStrings, options, footprint, URL, t
       };
     fetch(URL, requestOptions)
       .then(response => {
+        /* IF RESULT IN FORM: */
         response.json().then(calculatedFootprint => {
           setResult(calculatedFootprint)
-          increaseIndex()
+          setCurrentObject(resultObjects[0])
         })
-        //window.location.href = response.url
+        /* IF RESULT ON RESULT PAGE: */
+        // window.location.href = response.url
       })
       .catch(error => {
         console.log("Something went wrong, trying again.", error);
       })
   }
 
-  /**
-   * Saves the answer given
-   * Does some checks and saves to the footprint object 
-   */
-  function saveAnswer(givenAnswer) {
-    if (questionKeys[currentIndex] === "car_distance"){
-      footprint[questionKeys[currentIndex].concat("_week_answer")] = givenAnswer
-    } else {
-      footprint[questionKeys[currentIndex].concat("_answer")] = givenAnswer
-    }
-  }
-
-  /**
-   * Sets the index for next question
-   * Increases with at least 1, but if next question is not to be used, the index increases again
-   */
-  function increaseIndex(){
-    let nextIndex = currentIndex
-    do{
-      nextIndex++;
-    } while((calculator[(questionKeys[nextIndex]).concat("_options")] !== undefined 
-      && !calculator[(questionKeys[nextIndex]).concat("_options")])
-      && !resultKeys.includes(questionKeys[nextIndex]));    
-    setCurrentIndex(nextIndex)
-    return nextIndex
-  }
-  
-  /** 
-   * Called when changing to a previous question, decreases the index, if question is undefined decrease again
-   */
-  function decreaseIndex(){
-    let nextIndex = currentIndex
-    do{
-      nextIndex--
-    } while(calculator[(questionKeys[nextIndex]).concat("_options")] !== undefined 
-    && !calculator[(questionKeys[nextIndex]).concat("_options")]);    
-    setCurrentIndex(nextIndex)
-    return nextIndex
-  }
-
   /**
    * Called on go back-button, loads the previous question by decreasing index and setting the question and options
    */
   function onGoBack(){
-    let previousQuestionIndex = decreaseIndex()
-    if(!resultKeys.includes(questionKeys[previousQuestionIndex]))
-      setQuestion(previousQuestionIndex)
+    let newIndex = currentIndex - 1
+    if(newIndex < questionObjects.length && questionObjects[newIndex].questionKey === "car_distance" && footprint["car_type_answer"] === "no_car")
```

Comment:
```suggestion
    if(newIndex < questionObjects.length 
    && questionObjects[newIndex].questionKey === "car_distance" 
    && footprint["car_type_answer"] === "no_car")
```
Jag är inte emot denna lösning, men var inget jag tänkte på!

Reply:
Ser att det görs på andra ställen i samma fil